### DISCUSSION TITLE
Build Process takes too long

### DISCUSSION THREAD

**Discussion by Halembek** [Envek/aws-sam-typescript-layers-example#10] [General]
Hey guys thanks for this great template! I was using webpack to use typescript with AWS SAM until I found your template here. 
I was able to adapt the template to the project and managed to run it, unfortunately the building process takes too long just for one lambda (about 10 mins ), do you have any recommendations to improve the building process time ?

**Comment by Envek** [comment]
Hey!

That's weird because I've heard opposite feedback. when things have sped up a lot (especially when you have dozens of functions (see https://github.com/aws/aws-sam-cli/issues/2626#issuecomment-782584090 for example).

What runtime dependencies do you have in your `package.json`? Is there something heavy with native extensions that can be very slow inside Docker on some systems (as SAM CLI uses containers)?

Sadly, it is very hard to debug build process, please complain for that in https://github.com/aws/aws-sam-cli/issues/2006
